src/main/java/pl/javastart/task/Person.java

```diff
@@ -0,0 +1,80 @@
+package pl.javastart.task;
+
+public class Person {
+    private String firstName;
+    private String lastName;
+    private int age;
+    private String pesel;
+
+    public Person(String firstName, String lastName, int age, String pesel) throws NameUndefinedException, IncorrectAgeException {
+        if (firstName == null) {
+            this.setFirstName(null);
+        }
+        if (lastName == null) {
+            this.setLastName(null);
+        }
+        if (firstName.length() < 2) {
+            this.setFirstName("");
+        }
+        if (lastName.length() < 2) {
+            this.setLastName("");
+        }
+        if (age < 1) {
+            this.setAge(-1);
+        }
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.age = age;
+        this.pesel = pesel;
```

Comment:
Te ify nie mają za dużo sensu. Jeśli od razu uruchomisz np. `this.setFirstName(firstName)` to efekt będzie przecież ten sam. Po prostu usuń te bezpośrednie przypisania niżej.